### DISCUSSION TITLE
Changing fields in reports that are generated

### DISCUSSION THREAD

**Discussion by Xitija** [Sunbird-Obsrv/Community#98] [Query] [answered]
Hello @manjudr,

There are a few fields in the progress report that I do not require, and I want to remove them
The fields are highlighted in below picture.

![unwantedfields](https://github.com/Sunbird-Obsrv/Community/assets/65657373/91f3a479-5979-41cb-b3ea-347b4614fb15)

I need to add a field with name of user in this report, how can it be done?
How can I remove or add them during report generation?

cc : @Renuka-S @komalm @Nithin-ns24 @sravani-64 @vinod303 @gandham-santhosh

**Comment by manjudr** [comment]
@Xitija - This requires a few modifications in the report job.  Kindly revise this section to omit the unnecessary columns.
https://github.com/Sunbird-Ed/sunbird-data-products/blob/master/data-products/src/main/scala/org/sunbird/analytics/exhaust/collection/ProgressExhaustJob.scala#L47-L50

**Reply by komalm** [reply]
@manjudr Thank you! it's work.

Can you also help me in the following case - 

We want to show the user's first and last name in the same column (it's fine if we can show them in distinct columns). We do not wish to reveal the username.

Could you please advise if this is possible?

**Reply by manjudr** [reply]
@komalm  -  The job generates the username just by contacting the both `first name` and `last name` fields.  However, How does this help in username security? 
Refer to this code - https://github.com/Sunbird-Ed/sunbird-data-products/blob/master/data-products/src/main/scala/org/sunbird/analytics/exhaust/collection/BaseCollectionExhaustJob.scala#L288-L290

**Reply by manjudr** [reply, marked as answer]
@komalm 
If it is mandatory to show them as individual columns then we have to make the changes in the below files. 

1. Include both first name and last name properties and exclude the username property as part of the list here. 
 https://github.com/Sunbird-Ed/sunbird-data-products/blob/master/data-products/src/main/scala/org/sunbird/analytics/exhaust/collection/UserInfoExhaustJob.scala#L20-L21
